compiler/rustc_lint/src/late.rs

```diff
@@ -422,7 +422,9 @@ fn late_lint_crate<'tcx>(tcx: TyCtxt<'tcx>) {
         .into_iter()
         .filter(|pass| {
             let lints = (**pass).get_lints();
-            !lints.iter().all(|lint| lints_that_dont_need_to_run.contains(&LintId::of(lint)))
+            // Lintless passes are always in
+            lints.is_empty()
+                || !lints.iter().all(|lint| lints_that_dont_need_to_run.contains(&LintId::of(lint)))
```

Comment:
This condition is quite hard to read now. I suggest to store the second part in a variable:

```suggestion
            let all_lints_allowed = lints.iter().all(|lint| lints_that_dont_need_to_run.contains(&LintId::of(lint)));
            // `LintPass`es without lints always run
            lints.is_empty() || !all_lints_allowed
```

Reply:
Philipp wait because this PR doesn't handle the whole second part of your comment. As making a sync is a hard task, I assumed you wouldn't want to redo it. I'm currently testing making CTFE a lintless pass.

Reply:
Yeah, I can only do the sync after this is merged and got into nightly.

This was just a general note on this. I still like the variable name over the comment. But just a NIT, so feel free to keep either.